### PR TITLE
Support for OpenMined/PySyft-TensorFlow#11

### PR DESCRIPTION
**Pull request by yanndupis**
Hey, 

This PR add an `object_type` attribute to `ObjectPointer`. The goal is to be able to know if PointerTensor is pointing to a tf.Tensor or tf.Variable when [creating a wrapper](https://github.com/OpenMined/PySyft-TensorFlow/blob/7b66ffe9b9995ef8a38a7cc655b7908293952979/syft_tensorflow/hook/hook.py#L306) around it. 

Thanks,

### CODE REVIEW

File: syft/generic/pointers/object_pointer.py [hunk 1]
```diff
@@ -62,13 +64,18 @@ def __init__(
                 .grad. Note the string can be a chain (i.e., .child.child.child or
                 .grad.child.child). Defaults to None, which means don't point to any attr,
                 just point to then object corresponding to the id_at_location.
+            object_type: An option FrameworkObjectType to specify the object type which should
```
Comment (jvmancuso) [inline syft/generic/pointers/object_pointer.py]: *optional

File: syft/generic/pointers/pointer_tensor.py [hunk 2]
```diff
@@ -81,6 +83,10 @@ def __init__(
             tags: an optional set of strings corresponding to this tensor
                 which this tensor should be searchable for.
             description: an optional string describing the purpose of the tensor.
+            object_type: An option FrameworkObjectType to specify the object type which should
```
Comment (jvmancuso) [inline syft/generic/pointers/pointer_tensor.py]: *optional

File: syft/generic/pointers/pointer_tensor.py [hunk 3]
```diff
@@ -207,6 +215,10 @@ def create_pointer(
             local_autograd: Use autograd system on the local machine instead of PyTorch's
                 autograd on the workers.
             preinitialize_grad: Initialize gradient for AutogradTensors to a tensor.
+            object_type: An option FrameworkObjectType to specify the object type which should
```
Comment (jvmancuso) [inline syft/generic/pointers/pointer_tensor.py]: *optional

File: syft/generic/pointers/object_pointer.py (hunk 1 above)
Comment (yanndupis) [reply syft/generic/pointers/object_pointer.py]: ah thanks!

**Comment by LaRiffle** [comment]
Hey, I wonder if you can leverage what has been done in PyTorch to deal with Tensors & Parameters (which used to be called Variables)
We also create different pointers/wrappers when sending one or another.
Any thoughts?

**Comment by jvmancuso** [comment]
@LaRiffle can you point to where this happens?

**Comment by LaRiffle** [comment]
Will this PR be replaced by #2651 ?

**Comment by yanndupis** [comment]
Thanks for reaching out @LaRiffle. Correct, we will use the approach in #2651 instead. Let's close this pr.

**Pull request closed without merging**